flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/util/TempDirUtils.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.util;
+
+import javax.annotation.Nonnull;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.UUID;
+
+/**
+ * Util class for implementing common functions in junit4 {@link org.junit.rules.TemporaryFolder},
+ * like newFolder() and newFile() for junit5 {@link org.junit.jupiter.api.io.TempDir}.
+ */
+public class TempDirUtils {
+
+    private TempDirUtils() {}
+
+    public static Path newFolderIn(@Nonnull Path baseTempPath) {
```

Comment:
I'm not really sold on these tbh. From what I've seen most tests just to weird stuff making this often unnecessary. Explicitly defining the temp dirs for tests and passing them around in my mind makes it easier to understand what is actually done.